modules/integration-tests/src/test/java/org/wso2/carbon/apimgt/rest/integration/tests/store/APISubscriptionTestCaseIT.java

```diff
@@ -0,0 +1,179 @@
+                ("user1"), AMIntegrationTestConstants.DEFAULT_SCOPES).buildClient(SubscriptionCollectionApi.class);
+        SubscriptionList subscriptionList = subscriptionCollectionApi.subscriptionsGet("", application
+                .getApplicationId(), "", 0, 10, "");
+        Assert.assertNotNull(subscriptionList);
+        Assert.assertEquals(subscriptionList.getCount().intValue(), 1);
+        Subscription subscription = subscriptionList.getList().get(0);
+        Assert.assertEquals(subscription.getApiIdentifier(), this.subscription.getApiIdentifier());
+        Assert.assertEquals(subscription.getPolicy(), this.subscription.getPolicy());
+    }
+
+    @Test(dependsOnMethods = "testCreateSubscription")
+    public void testGetAllSubscriptionsByAPIId() throws AMIntegrationTestException {
+        SubscriptionCollectionApi subscriptionCollectionApi = TestUtil.getStoreApiClient("user1", TestUtil.getUser
+                ("user1"), AMIntegrationTestConstants.DEFAULT_SCOPES).buildClient(SubscriptionCollectionApi.class);
+        SubscriptionList subscriptionList = subscriptionCollectionApi.subscriptionsGet(subscription.getApiIdentifier
+                (), "", "", 0, 10, "");
+        Assert.assertNotNull(subscriptionList);
+        Assert.assertEquals(subscriptionList.getCount().intValue(), 1);
+        Subscription subscription = subscriptionList.getList().get(0);
+        Assert.assertEquals(subscription.getApplicationId(), this.subscription.getApplicationId());
+        Assert.assertEquals(subscription.getPolicy(), this.subscription.getPolicy());
+    }
+
+    @Test(dependsOnMethods = "testCreateApplication")
+    public void testCreateSubscriptionWIthNonExistingPolicy() {
+        String apiId = TestUtil.getApi("baseapi1").getId();
+        String applicationId = application.getApplicationId();
+        Subscription subscriptionNegative = new Subscription().apiIdentifier(apiId).applicationId(applicationId)
+                .policy("Bronze1");
+        try {
+            subscriptionIndividualApi.subscriptionsPost(subscriptionNegative);
+            Assert.fail();
+        } catch (Exception e) {
+            if (e.getCause() instanceof RestAPIException) {
+                RestAPIException restAPIException = (RestAPIException) e.getCause();
+                Error error = restAPIException.getError();
+                Assert.assertEquals(error.getMessage(), "Policy Not found");
+                Assert.assertNotNull(error.getMoreInfo());
+                Map<String, String> moreInfo = error.getMoreInfo();
+                Assert.assertEquals(moreInfo.get("API_ID"), apiId);
+                Assert.assertEquals(moreInfo.get("APPLICATION_ID"), applicationId);
+                Assert.assertEquals(moreInfo.get("TIER"), "Bronze1");
+            } else {
+                Assert.fail();
+            }
+        }
+    }
+
+    @AfterClass
+    public void destroy() {
```

Comment:
Any reason why we are doing this explicitly since we are anyway dropping the tables after each test run?